lib/puppet/node/facts.rb

```diff
@@ -116,8 +116,25 @@ def add_timestamp
     @timestamp = Time.now
   end
 
+  def to_yaml
+    facts_to_display = Psych.parse_stream(YAML.dump(self))
+    quote_special_strings(facts_to_display)
+  end
+
   private
 
+  def quote_special_strings(fact_hash)
+    fact_hash.grep(Psych::Nodes::Scalar).each do |node|
+      next unless node.value =~ /:/
+
+      node.plain  = false
+      node.quoted = true
+      node.style  = Psych::Nodes::Scalar::DOUBLE_QUOTED
```

Comment:
I'm not super familiar with what these options mean, could you update the commit summary to describe how it changes the behavior?

Reply:
Added